apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server_test.py

```diff
@@ -0,0 +1,456 @@
+
+  if expect_metadata_header:
+    check_header(headers, url, common.metadata_flavor, common.metadata_flavor_google)
+  else:
+    assert common.metadata_flavor.lower() not in headers
+
+  check_header(headers, url, common.server, common.server_value)
+  check_header(headers, url, common.content_type, expected_content_type)
+  
+  contents = response.read()
+  assert expected == contents
+
+def expect_content(path, expected, expected_content_type, expect_metadata_header=True):
+  expect_content_absolute(server_address + '/computeMetadata/v1/' + path,
+                          expected,
+                          expected_content_type,
+                          expect_metadata_header)
+
+def expect_error(path, expected_code, metadata_flavor_value=common.metadata_flavor_google):
+  url = server_address + '/computeMetadata/v1/' + path
+  request = urllib2.Request(url,
+                            headers={common.metadata_flavor :
+                                     metadata_flavor_value} if metadata_flavor_value else {})
+  try:
+    response = urllib2.urlopen(request)
+    # This will always fail but I'm asserting so the response appears in the test results
+    assert response.read() is None
+  except urllib2.HTTPError as e:
+    check_header(e.headers, url, common.metadata_flavor, common.metadata_flavor_google)
+    check_header(e.headers, url, common.server, common.server_value)
+    check_header(e.headers, url, common.content_type, common.content_type_html)
+    assert e.code == expected_code
+
+def check_path(path, expected, default='text'):
+  expected_json = json.dumps(expected, separators=(',', ':'))
+  if isinstance(expected, list):
+    expected_text = ''.join(s + '\n' for s in expected)
+  else:
+    expected_text = unicode(expected)
+    
+  expect_content(path,
+                 expected_text if default == 'text' else expected_json,
+                 common.content_type_text if default == 'text' else common.content_type_json)
+  expect_content(path + "?alt=text", expected_text, common.content_type_text)
+  expect_content(path + "?alt=json", expected_json, common.content_type_json)
+
+def check_dir(path, expected):
+  expect_content(path, expected, common.content_type_text)
+  if path:
+    expect_content(path + '/', expected, common.content_type_text)
+
+def check_dir_absolute(url, expected):
+  expect_content_absolute(url, expected, common.content_type_text)
+  expect_content_absolute(url + '/', expected, common.content_type_text)
+
+def check_dir_recursive(path, expected, test_trailing_slash=True, use_recursive_field=True, expect_metadata_header=True):
+  query = '?recursive=true' if use_recursive_field else ''
+  expect_content(path + query, expected, common.content_type_json, expect_metadata_header)
+  if test_trailing_slash:
+    expect_content(path + '/' + query, expected, common.content_type_json, expect_metadata_header)
+
+  data = json.loads(expected, object_pairs_hook=OrderedDict)
+
+  def convertToText(prefix, o):
```

Comment:
This is already part of the data types as prepareRecursiveResultAsText. However, it would make me a little nervous to have the tests share this logic with the emulator since a bug introduced in the shared logic in the emulator would cause the same bug to be introduced for the testing code, and since that would make the values compare equal, we would never detect it. So I'm leaning towards keeping the manual formatting logic separate here, but if you feel strongly about it, I'll share it and add separate tests for that as you suggest.